### PR TITLE
36907 fix WP_Query obey parameters for sticky posts

### PR DESCRIPTION
**Pull request by uday-kokitkar**
Trac ticket: 

XWP: https://xwp-co.atlassian.net/browse/INITS-370
WP: https://core.trac.wordpress.org/ticket/36907

**Note that** there is a bug related to Sticky posts in WordPress core and hence the unit tests would fail in this PR.
Bug: Sticky posts do not appear at the top.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/query/stickies.php [hunk 1]
```diff
@@ -48,11 +48,27 @@ public function test_stickies_should_be_included_when_is_home_is_true() {
...
+
 		$this->assertSame( self::$posts[2], $q->posts[0]->ID );
 		$this->assertSame( self::$posts[8], $q->posts[1]->ID );
 		$this->assertSame( self::$posts[14], $q->posts[2]->ID );
 	}
 
+	public function get_query_from_filter( $results, $query ) {
+		$sticky_posts = get_option( 'sticky_posts' );
+		if ( $query->query_vars['post__in'] === $sticky_posts ) {
+			$this->query_vars = $query->query_vars;
```
Comment (spacedmonkey) [inline tests/phpunit/tests/query/stickies.php:71]: This property should be defined in the class.

**Comment by spacedmonkey** [comment]
One nit pic, but this is nearly there.

File: tests/phpunit/tests/query/stickies.php (hunk 1 above)
Comment (uday-kokitkar) [reply tests/phpunit/tests/query/stickies.php:71]: @spacedmonkey are you referring to `$this->query_vars` this?
I now realized that we have not used that variable anywhere in the class. And I'm too afraid to ask this question now. 
Should we use it here? https://github.com/xwp/wordpress-develop/blob/4d0692d93cf56d48e68e5ed749535246864be4c6/tests/phpunit/tests/query/stickies.php#L55-L57

File: tests/phpunit/tests/query/stickies.php [hunk 2]
```diff
@@ -48,11 +48,27 @@ public function test_stickies_should_be_included_when_is_home_is_true() {
 
 		$q = $GLOBALS['wp_query'];
 
+		add_filter( 'posts_pre_query', array( $this, 'get_query_from_filter' ), 10, 2 );
+		$sticky_posts = get_option( 'sticky_posts' );
+
+		$this->assertSame( ( count( $sticky_posts ) + $q->query_vars['posts_per_page'] ), $q->post_count ); // The total posts should be equal to sticky posts + posts_per_page.
+		$this->assertFalse( $q->query_vars['suppress_filters'] );
+		$this->assertTrue( $q->query_vars['update_post_meta_cache'] );
+		$this->assertTrue( $q->query_vars['update_post_term_cache'] );
```
Comment (spacedmonkey) [inline tests/phpunit/tests/query/stickies.php]:
```suggestion
		$this->assertSame( $q->query_vars['suppress_filters'], $this->query_vars['suppress_filters'] );
		$this->assertSame( $q->query_vars['update_post_meta_cache'], $this->query_vars['suppress_filters'] );
		$this->assertSame( $q->query_vars['update_post_term_cache'], $this->query_vars['suppress_filters'] );
```

File: tests/phpunit/tests/query/stickies.php (hunk 1 above)
Comment (spacedmonkey) [reply tests/phpunit/tests/query/stickies.php:71]: Oh, you are right, this is what the test should look like - https://github.com/xwp/wordpress-develop/pull/6/files#r831931039

**Pull request closed without merging**